### PR TITLE
Add unified interface test utils

### PR DESCRIPTION
**Pull request by sharanry**
#18 

Proposes a unified interface testing utilities similar to the ones added by @willtebbutt  in https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/pull/159.

### CODE REVIEW

File: test/test_utils.jl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    end
+end
+
+"""
+    test_interface(
+        lik,
+        k::Kernel,
+        x::AbstractVector;
+        functor_args=(),
+    )
```
Comment (willtebbutt) [inline test/test_utils.jl:41]:
```suggestion
    test_interface(lik, k::Kernel, x::AbstractVector; functor_args=())
```

File: test/test_utils.jl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    x::AbstractVector;
+    kwargs...
+)
+    test_interface(
+        Random.GLOBAL_RNG,
+        lik,
+        k,
+        x;
+        kwargs...
+    )
```
Comment (willtebbutt) [inline test/test_utils.jl:59]:
```suggestion
    test_interface(Random.GLOBAL_RNG, lik, k, x; kwargs...)
```

File: test/test_utils.jl [hunk 3]
```diff
@@ -0,0 +1,60 @@
+function test_interface(
+    rng::AbstractRNG,
+    lik,
+    k::KernelFunctions.Kernel,
+    x::AbstractVector;
+    functor_args=(),
+)
```
Comment (willtebbutt) [inline test/test_utils.jl:7]:
```suggestion
function test_interface(
    rng::AbstractRNG, lik, k::Kernel, x::AbstractVector; functor_args=(),
)
```

File: test/test_utils.jl [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+"""
+    test_interface(
+        lik,
+        k::Kernel,
+        x::AbstractVector;
+        functor_args=(),
+    )
+
+This function provides unified method to check the interface of the various likelihoods
```
Comment (willtebbutt) [inline test/test_utils.jl:43]: This probably needs a more careful description of `functor_args` -- it's not completely clear to me what they need to be.

**Review by willtebbutt — COMMENTED** [review]
This is looking good. Just a couple of style-related things.

We also need a patch version bump :)

**Comment by sharanry** [comment]
> We also need a patch version bump :)

Since we haven't registered the package and we aren't making any releases yet, are the patch bumps useful?

Also should we register once this is merged?

**Comment by theogf** [comment]
Haha no patch bump, I think @willtebbutt just forgot it was not registered! We need to leave it at 0.1.0 or it will not be automatically merged

**Comment by willtebbutt** [comment]
🤦 @theogf is exactly correct - I keep forgetting that this package isn't registered yet...

**Comment by sharanry** [comment]
> facepalm @theogf is exactly correct - I keep forgetting that this package isn't registered yet...

Me too 😆. Should we merge this and register now that we set up the tests?

**Comment by willtebbutt** [comment]
Sounds good to me.

**Pull request merged**